binaries/daemon/src/spawn.rs

```diff
@@ -301,14 +301,49 @@ pub async fn spawn_node(
                     cmd
                 }
             } else if python_operators.is_empty() && other_operators {
-                let mut cmd = tokio::process::Command::new(
-                    std::env::current_exe().wrap_err("failed to get current executable path")?,
-                );
-                cmd.arg("runtime");
-                cmd
+                let current_exe =
+                    std::env::current_exe().wrap_err("failed to get current executable path")?;
+                let mut file_name = current_exe.clone();
+                file_name.set_extension("");
+                let file_name = file_name
+                    .file_name()
+                    .and_then(|s| s.to_str())
+                    .context("failed to get file name from current executable")?;
+
+                // Check if the current executable is a dora binary
+                if file_name.ends_with("dora") {
```

Comment:
Relying on the executable name seems a bit hacky. For example, users might want to install multiple versions of dora at the same time and rename the executables for that. For example, they might have a `dora-0.3.9` and a `dora-0.3.11` executable.

I think the approach of #940 is more robust. By keeping track of the call site in `DoraCommand`, we know for sure whether `dora` was invoked as a binary or from a Python environment. (We just need to fix the command for Python environments in https://github.com/dora-rs/dora/pull/940/files#diff-30db22706909157739f838586f6c2282008e9fa502c3e0d54a818cafdec5893fR2213-R2227 . I think we could use the ` "-c", format!("import dora; dora.start_runtime() # {}", node.id).as_str()` from this PR for that.)

Reply:
What about we just check for python and if it's not python. we fall back to the current exe.

There is actually a lot of things that could go wrong with https://github.com/dora-rs/dora/pull/940/files#diff-30db22706909157739f838586f6c2282008e9fa502c3e0d54a818cafdec5893fR2213-R2227 as there is no guarantee that the python used for the cli is the same that the one from the current env in which we want to spawn the runtime node. In which case, using #940 is going to be extremely difficult to debug.

Reply:
I'm going to modify the current code to be able to spawn the current interpreter, this means we will be able to install shared-library from pip.

Reply:
Yes, the python spawn command of #940 is not correct. What i meant is that we could use the `DoraCommand` approach to avoid the executable name check. My idea was that we keep track of whether we enter the dora code from the Rust `main` function or from one of the Python functions (`run` or `py_main`). This way, we don't need to check the executable file name since we already know whether it's a Rust executable or Python-based.